internal/handler/command.go

```diff
@@ -403,8 +482,17 @@ func parseWriteParams(profileName string, ros []contract.ResourceOperation, para
 	return result, nil
 }
 
-func parseParams(params string) (paramMap map[string]string, err error) {
-	err = json.Unmarshal([]byte(params), &paramMap)
+func parseParams(params string, r *http.Request) (paramMap map[string]string, err error) {
+
+	// Check the header value
+	if r.Header.Get(clients.ContentType) == clients.ContentTypeCBOR {
+		// Handle Content-type CBOR and populate paramMap
+		err = codec.NewDecoderBytes([]byte(params), &codec.CborHandle{}).Decode(&paramMap)
+	} else if r.Header.Get(clients.ContentType) == clients.ContentTypeJSON {
+		// Handle Content-type JSON and populate paramMap
+		err = json.Unmarshal([]byte(params), &paramMap)
+	}
```

Comment:
Can we do this in a switch statement? That way we are explicitly handling each content type, and we can have a default case which either returns an error or assumes JSON with a comment stating that is our intent.

I do not know if you meant to use the Readers, which I commented was not being used,  here rather than directly creating and using unmarshalers, but I think using the Readers would be better since we can simply add functionality for additional content types by creating new readers rather than adding more switch/if-else statements here. This would better follow the open-closed principal.

Reply:
> Can we do this in a switch statement? That way we are explicitly handling each content type, and we can have a default case which either returns an error or assumes JSON with a comment stating that is our intent.
> 
> I do not know if you meant to use the Readers, which I commented was not being used, here rather than directly creating and using unmarshalers, but I think using the Readers would be better since we can simply add functionality for additional content types by creating new readers rather than adding more switch/if-else statements here. This would better follow the open-closed principal.

@AnthonyMBonafide Definitely, I will follow up, in particular regarding the following point in your comment above:

> I do not know if you meant to use the Readers, which I commented was not being used, here rather than directly creating and using unmarshalers, but I think using the Readers would be better since we can simply add functionality for additional content types by creating new readers rather than adding more switch/if-else statements here. This would better follow the open-closed principal.